x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/ConcatFunctionProcessor.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.sql.SqlIllegalArgumentException;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.runtime.BinaryProcessor;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.runtime.Processor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ConcatFunctionProcessor extends BinaryProcessor {
+
+    public static final String NAME = "cb";
+
+    public ConcatFunctionProcessor(Processor source1, Processor source2) {
+        super(source1, source2);
+    }
+    
+    public ConcatFunctionProcessor(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    protected Object doProcess(Object source1, Object source2) {
+        return doProcessInScripts(source1, source2);
+    }
+    
+    /**
+     * Used in Painless scripting
+     */
+    public static Object doProcessInScripts(Object source1, Object source2) {
+        if (source1 == null) {
+            return source2;
+        }
+        if (source2 == null) {
+            return source1;
+        }
+        if (!(source1 instanceof String || source1 instanceof Character)) {
+            throw new SqlIllegalArgumentException("A string/char is required; received [{}]", source1);
+        }
+        if (!(source2 instanceof String || source2 instanceof Character)) {
+            throw new SqlIllegalArgumentException("A string/char is required; received [{}]", source2);
+        }
+
+        return (source1 instanceof Character ? source1.toString() : (String) source1)
```

Comment:
I'd just call `toString` on both args. If they happen to be `String`s then it'll just return the string again. That is cheaper than checking the `instanceof` I think.